ext/session/session.c

```diff
@@ -485,24 +485,23 @@ PHPAPI int php_session_valid_key(const char *key) /* {{{ */
 /* }}} */
 
 
-static void php_session_gc(void) /* {{{ */
+static zend_long php_session_gc(zend_bool immediate) /* {{{ */
 {
 	int nrand;
+	int nrdels = -1;
 
 	/* GC must be done before reading session data. */
 	if ((PS(mod_data) || PS(mod_user_implemented)) && PS(gc_probability) > 0) {
-		int nrdels = -1;
-
 		nrand = (int) ((float) PS(gc_divisor) * php_combined_lcg());
+		if (immediate) {
+			PS(mod)->s_gc(&PS(mod_data), PS(gc_maxlifetime), &nrdels);
+			return (zend_long)nrdels;
+		}
 		if (nrand < PS(gc_probability)) {
 			PS(mod)->s_gc(&PS(mod_data), PS(gc_maxlifetime), &nrdels);
-#ifdef SESSION_DEBUG
```

Comment:
hmm, I thought there were others ... but this seems to be the only one ... 

I guess it can go, though it doesn't really need to be part of this PR/RFC.